src/NServiceBus.Core/Routing/EndpointInstances.cs

```diff
@@ -34,20 +37,24 @@ public void AddDynamic(Func<EndpointName, Task<IEnumerable<EndpointInstance>>> d
         /// <summary>
         /// Adds static information about an endpoint.
         /// </summary>
-        /// <param name="endpoint">Name of the endpoint.</param>
         /// <param name="instances">A static list of endpoint's instances.</param>
-        public void Add(EndpointName endpoint, params EndpointInstance[] instances)
+        public void Add(params EndpointInstance[] instances) => Add((IEnumerable<EndpointInstance>)instances);
+
+        /// <summary>
+        /// Adds static information about an endpoint.
+        /// </summary>
+        /// <param name="instances">A static list of endpoint's instances.</param>
+        public void Add(IEnumerable<EndpointInstance> instances)
         {
-            Guard.AgainstNull(nameof(endpoint), endpoint);
-            if (instances.Length == 0)
+            if (!instances.Any())
             {
                 throw new ArgumentException("The list of instances can't be empty.", nameof(instances));
             }
-            if (instances.Any(i => i.Endpoint != endpoint))
+            var endpointsByName = instances.GroupBy(i => i.Endpoint);
+            foreach (var instanceGroup in endpointsByName)
             {
-                throw new ArgumentException("At least one of the instances belongs to a different endpoint than specified in the 'endpoint' parameter.", nameof(instances));
```

Comment:
@SzymonPobiega what do you mean by covered? since the EndpointInstance contains the associated EndpointName, the instances don't need to belong to the same EndpointName, the grouping does split them up internally.